spec/index.md

```diff
@@ -780,17 +780,31 @@ Given a query (the `content` map from the request body) `Q`, a response `R`, and
               status: "replied",
               reply: R.reply,
               timestamp: T,
-              request_id: hash_of_map(Q)}))
+              request_id: hash_of_map({
+                request_type: "query",
+                sender: Q.sender,
+                nonce: Q.nonce,
+                canister_id: Q.canister_id,
+                method_name: Q.method_name,
+                arg: Q.arg
+              })}))
```

Comment:
Would it make sense to use some notation which makes it clear that all fields of `Q` except `ingress_expiry` are used?

Reply:
I've given it a try. Could you please take a look?

Reply:
Thanks a lot. Looks good to me. Maybe we could consider using the `\` (i.e., set minus operator) instead of `-` but I have no strong opinion here as the explanation makes it clear what we mean anyways.

Reply:
Sure, I haven't seen either `\` or `-` in the current spec so changed to `\`.